application/org.openjdk.jmc.joverflow.ui/src/main/java/org/openjdk/jmc/joverflow/ui/swt/BreadcrumbItem.java

```diff
@@ -0,0 +1,319 @@
+public class BreadcrumbItem extends Item {
+	private static final int PADDING = 4;
+	private static final int ARROW_WIDTH = 4;
+
+	private Breadcrumb parent;
+
+	private Color background = null;
+	private Color foreground = null;
+	private Font font = null;
+
+	private Rectangle bounds = null;
+	private Rectangle textBounds = null;
+
+	// to be disposed
+	private Color darkenBackground = null;
+	private Color lighterForeground = null;
+
+	/**
+	 * Constructs a new instance of this class and inserts it into the parent breadcrumb. The item
+	 * is inserted as the last item maintained by its parent.
+	 * 
+	 * @param parent
+	 *            a breadcrumb control which will be the parent of the new instance (cannot be null)
+	 * @param style
+	 *            the style of control to construct
+	 */
+	public BreadcrumbItem(Breadcrumb parent, int style) {
+		super(Breadcrumb.checkNull(parent), style);
+
+		this.parent = parent;
+		parent.createItem(this);
+	}
+
+	/* package-private */ void paintItem(GC gc, Rectangle bounds) {
+		Color bg = gc.getBackground();
+		Color fg = gc.getForeground();
+		Font font = gc.getFont();
+
+		gc.setFont(getFont());
+
+		textBounds = new Rectangle(bounds.x + PADDING + ARROW_WIDTH, bounds.y + PADDING, bounds.width, bounds.height);
+		Point textExtent = gc.textExtent(getText());
+		textBounds.width = textExtent.x;
+		textBounds.height = textExtent.y;
+
+		bounds = new Rectangle(bounds.x, bounds.y, textBounds.width + 2 * PADDING + 2 * ARROW_WIDTH,
+				textBounds.height + 2 * PADDING);
+		this.bounds = bounds;
+
+		gc.setForeground(getBackground());
+		gc.setBackground(getDarkenBackground());
+		gc.fillGradientRectangle(bounds.x, bounds.y, bounds.width, bounds.height, true);
+
+		gc.setBackground(getBackground());
+		gc.setForeground(getForeground());
+		int[] polygon = new int[] {bounds.x, bounds.y, //
+				bounds.x + ARROW_WIDTH, bounds.y + bounds.height / 2, //
+				bounds.x, bounds.y + bounds.height, //
+		};
+		gc.fillPolygon(polygon);
+
+		polygon = new int[] {bounds.x + bounds.width - ARROW_WIDTH, bounds.y, //
+				bounds.x + bounds.width, bounds.y, //
+				bounds.x + bounds.width, bounds.y + bounds.height, //
+				bounds.x + bounds.width - ARROW_WIDTH, bounds.y + bounds.height, //
+				bounds.x + bounds.width, bounds.y + bounds.height / 2, //
+		};
+		gc.fillPolygon(polygon);
+
+		gc.setForeground(getLighterForeground());
+		polygon = new int[] {bounds.x, bounds.y, //
+				bounds.x + bounds.width - ARROW_WIDTH, bounds.y, //
+				bounds.x + bounds.width, bounds.y + bounds.height / 2, //
+				bounds.x + bounds.width - ARROW_WIDTH, bounds.y + bounds.height, //
+				bounds.x, bounds.y + bounds.height, //
+				bounds.x + ARROW_WIDTH, bounds.y + bounds.height / 2,};
+		gc.drawPolygon(polygon);
+
+		gc.setForeground(getForeground());
+		gc.drawText(getText(), textBounds.x, textBounds.y, true);
+
+		gc.setBackground(bg);
+		gc.setForeground(fg);
+		gc.setFont(font);
+	}
+
+	/* package-private */ Point getDimension(GC gc) {
```

Comment:
Nit: We usually don't call out things as being package private in comments. The Java language (and the IDE) does that for you. If you want to call out a design decision, explaining the decision in the comment may be more helpful. This is not a blocker - I'm okay with it, if you find it helpful.

Reply:
I saw this practice when studying the Eclipse SWT source. There is no important reason for explicitly calling out these methods/classes being`/* package-private */`. I thought it was more like a good-to-have sort of things.

I'll refrain from doing this in the future. Please let me know if you prefer removing these comments throughout this patch.

Reply:
Personally I'd prefer to have them removed.

Reply:
Or added as a proper comment with the reasoning for them being package private...

/*
 * This method is package private since ClassX needs to access the internal state without having to do a defensive copy every time, but we don't want to have it exposed in the public API since blahblahblah.
*/

Reply:
> Personally I'd prefer to have them removed.

All `/* package-private */` comments are removed in commit aaaab0b.